x/emissions/keeper/msg_server.go

```diff
@@ -221,6 +221,37 @@ func (ms msgServer) ProcessInferences(ctx context.Context, msg *state.MsgProcess
 	return &state.MsgProcessInferencesResponse{}, nil
 }
 
+// A tx function that accepts a list of forecasts and possibly returns an error
+// Need to call this once per forecaster per topic inference solicitation round because protobuf does not nested repeated fields
+func (ms msgServer) ProcessForecasts(ctx context.Context, msg *state.MsgProcessForecasts) (*state.MsgProcessForecastsResponse, error) {
```

Comment:
Should this come in a second tx? I thought it would be an opcional parameter in the Inference.